src/cli.cr

```diff
@@ -50,7 +50,7 @@ module Shards
         when "install"
           Commands::Install.run(path)
         when "list"
-          Commands::List.run(path)
+          Commands::List.run(path, args)
```

Comment:
This should pass on `args[1, -1]` like in the build command.

Reply:
Other commands don't take specific options, but arguments, I'd prefer the following to avoid pushing this parsing to command objects:

```crystal
class Command::List
  def self.run(@@tree = false)
  end
end

Command::List.new(tree: args.includes?("--tree"))
```

Reply:
I'm not sure how **you** would refactor the whole `Command::List` to do what you want though.

If we just call `Command::List.new(tree: args.includes?("--tree"))`, we lose the `path` that the abstract class expect to initialize, and well nothing works... I was not expecting to refactor all the CLI to just add `--tree`. I think it's a different PR. :sweat_smile:

Reply:
`path` should be there as first argument. @ysbaddaden probably just missed it.

Reply:
Alright, I think [that commit](https://github.com/crystal-lang/shards/pull/237/commits/95bd96a6f845ae25e4db84d153336323d284429b) is pretty much what @ysbaddaden was asking. :thinking: